lib/nexmo_developer/app/services/translator/file_translator.rb

```diff
@@ -0,0 +1,43 @@
+module Translator
+  class FileTranslator
+    attr_reader :doc_path
+
+    def initialize(doc_path)
+      @doc_path = doc_path
+    end
+
+    def translation_requests
+      I18n.available_locales.map do |locale|
+        Translator::TranslationRequest.new(locale: locale, frequency: frequency, path: doc_path)
+      end
+    end
+
+    def frontmatter
+      @frontmatter ||= YAML.safe_load(File.read("#{Rails.configuration.docs_base_path}/_documentation/en/#{doc_path}"))
+    end
+
+    def frequency
+      return frontmatter['translation_frequency'] if frontmatter['translation_frequency']
+
+      product_translation_frequency
+    end
+
+    def find_product
+      products = YAML.safe_load(File.open("#{Rails.configuration.docs_base_path}/config/products.yml"))
```

Comment:
this is just memoizing `@product`, we should memoize the whole method.
something like

```
def product
  @product ||= begin
     # code goes here...
  end
end
```